### ISSUE TITLE
Improvement: treat named pipes as the PIPE input is handled

### ISSUE THREAD

**Issue by 0bi-w6n-K3nobi** [allinurl/goaccess#1970]
Hello @allinurl

I would like to speak here about some requests for improvement.

Now that `GOACCES` handles each `LOG`, with its `HASH` information and others, it would be easy to add this improvement.

Again, I find it simpler to demonstrate through an example.
See you follow, the execution:
```
mkfifo fifo-a
mkfifo fifo-b
mkfifo fifo-c
cat LOG-a | awk ... >> fifo-a &
cat LOG-b | awk ... >> fifo-b &
cat LOG-c | awk ... >> fifo-c &
goaccess --outuput=. /all.html fifo-a fifo-b fifo-c
```

In this case, it is necessary to do the same treatment as the `PIPE` input ... For the most correctless processing.
I.E. --  identify the `LOG` as a named pipe and open it with the `O_NONBLOCK` flag.

**Comment by 0bi-w6n-K3nobi** [comment]
Ok. This make sense only in real-time.

Just as a note -- the challenges/requirements that must be met/considered:

+ change conditionals with `glog-> filename` and` glog->pipe` -- maybe use `glog->piping`?;
+ change the function `set_pipe_stdin` to also open the `LOGs` [named pipe] properly;
+ change logic to not close the `LOGs`;
+ `persist` and `ht_insert_last_parse` -- maybe use fifo inode?;

See you later.

**Comment by allinurl** [comment]
Thanks for suggesting this. I like the idea, I want to make sure I understand better the use case, could you please share what do you have in mind? and if there's a way we could do it right now, e.g., 

    tail -f /var/log/log-a -f /var/log/log-b | stdbuf -o0 awk -F" " '{print $1, $2}' | goaccess -

**Comment by 0bi-w6n-K3nobi** [comment]
Yeah. Of course can you made it.
And now, it may seem like it doesn't make sense.

But ...
_A simple case_ -- you think that **each** named pipe can be **output from a server** via `ssh`.
_And in the future_ -- when we have the **multi-thread** this will look more promising.

See you soon.

**Comment by allinurl** [comment]
I agree, I also see the use case where you could have different awk/sed filters.
